rust/src/adapter/lightstep.rs

```diff
@@ -0,0 +1,153 @@
+    fn handle_trace_event(&mut self, trace_evt: TraceEvent) -> Result<()> {
+        self.trace_events.push(trace_evt);
+        self.dump_traces()?;
+        Ok(())
+    }
+}
+
+impl LightstepAdapter {
+    /// Creates the Lightstep adapter and spawns a task for it.
+    /// This should ideally be created once per process of
+    /// your rust application.
+    pub fn create(config: LightstepConfig) -> AdapterHandle {
+        Self::spawn(Self {
+            config,
+            trace_events: vec![],
+        })
+    }
+
+    fn event_to_spans(
+        &self,
+        spans: &mut Vec<opentelemetry::proto::trace::v1::Span>,
+        event: Event,
+        parent_id: Vec<u8>,
+        trace_id: String,
+        meta: &Option<AdapterMetadata>,
+    ) -> Result<()> {
+        match event {
+            Event::Func(f) => {
+                let name = f.name.clone().unwrap_or("unknown-name".to_string());
+
+                let mut span =
+                    OtelFormatter::new_span(trace_id.clone(), parent_id, name, f.start, f.end);
+                let span_id = span.span_id.clone();
+                if let Some(m) = meta {
+                    if let AdapterMetadata::OpenTelemetry(m) = m {
+                        for entry in m.iter() {
+                            if let Some(v) = entry.value.int_value {
+                                OtelFormatter::add_attribute_i64_to_span(
+                                    &mut span,
+                                    entry.key.clone(),
+                                    v,
+                                )
+                            } else if let Some(v) = entry.value.string_value.clone() {
+                                OtelFormatter::add_attribute_string_to_span(
+                                    &mut span,
+                                    entry.key.clone(),
+                                    v,
+                                )
+                            }
+                        }
+                    }
+                }
+                spans.push(span);
+
+                for e in f.within.iter() {
+                    self.event_to_spans(
+                        spans,
+                        e.to_owned(),
+                        span_id.clone(),
+                        trace_id.clone(),
+                        &meta,
+                    )?;
+                }
+            }
+            Event::Alloc(a) => {
+                let mut span = OtelFormatter::new_span(
+                    trace_id,
+                    parent_id,
+                    "allocation".to_string(),
+                    a.ts,
+                    a.ts,
+                );
+                OtelFormatter::add_attribute_i64_to_span(
+                    &mut span,
+                    "amount".to_string(),
+                    a.amount.into(),
+                );
+                spans.push(span);
+            }
+            _ => {}
+        }
+        Ok(())
+    }
```

Comment:
same comment re: attaching the allocation as an attribute to the previous function in the spans.